src/domain.rs

```diff
@@ -151,11 +163,16 @@ impl Domain<Global> {
 }
 
 /// Generate a [`Domain`] with an entirely unique domain family.
+/// The generated family implements [`Singleton`],
 #[macro_export]
 macro_rules! unique_domain {
-    () => {
-        Domain::new(&|| {})
-    };
+    () => {{
+        struct UniqueFamily;
+        // Safety: UniqueFamily is only visible inside this scope,
+        // therefore no other Domain of that family can be constructed.
```

Comment:
"as nowhere else can construct an instance of `UniqueFamily` to pass to `Domain::new`."